### ISSUE TITLE
Circle test picks up release package installed from conda and not current branch esmvalcore

### ISSUE THREAD

**Issue by valeriupredoi** [ESMValGroup/ESMValCore#1063]
The [Circle test](https://app.circleci.com/pipelines/github/ESMValGroup/ESMValCore/4590/workflows/7680d683-a744-4b51-9b81-0edfddcbd966/jobs/21544) runs with the esmvalcore installed from conda at env creation point and is not overwritten by the <dev> package from the current branch under testing. This is not the desired behavior since `pip install -e .[test]` does not replace the conda install with the local pip install

**Comment by bouweandela** [comment]
Why do you think this is the case?

The [`esmvalgroup/esmvalcore:development`](https://hub.docker.com/layers/esmvalgroup/esmvalcore/development/images/sha256-adc263c4f0b4c3099d2353d9b66ce552cb672a46d3b624868088afa028e9b4f6?context=explore) docker container is built from the file [docker/Dockerfile.dev](https://github.com/ESMValGroup/ESMValCore/blob/master/docker/Dockerfile.dev). The test is run by starting this docker container and running the next commands in it:
https://github.com/ESMValGroup/ESMValCore/blob/c0cc45fde40e4b4f943cb47df262a6d183128a93/.circleci/config.yml#L33-L36
. Nowhere in these steps is there any mention of the `esmvalcore` conda package.

**Comment by valeriupredoi** [comment]
@bouweandela here's yer proof: [GA test](https://github.com/ESMValGroup/ESMValCore/pull/999/checks?check_run_id=2225962621) run in development on the branch using the actual branch codebase (doesn't fail, as I'd expect it) vs [Circle test](https://app.circleci.com/pipelines/github/ESMValGroup/ESMValCore/4593/workflows/8ba4bb9c-7c69-4b15-967c-1d26daab97a2/jobs/21559) that's failing because it is using either master or release codebase (I think it's the release since look where it picks up esmvalcore from: `/opt/conda/envs/esmvaltool/lib/python3.9/site-packages/esmvalcore`). If `test` is supposed to check the release codebase against new testing code (because that's what it's doing at the moment) then we should explain this, if not we need to fix it, I believe replacing `pip install -e .[test]` with `pip install -e .[develop]` will do the trick :+1:

**Comment by bouweandela** [comment]
> I think it's the release since look where it picks up esmvalcore from: /opt/conda/envs/esmvaltool/lib/python3.9/site-packages/esmvalcore

It gets installed there because this command
https://github.com/ESMValGroup/ESMValCore/blob/c0cc45fde40e4b4f943cb47df262a6d183128a93/.circleci/config.yml#L35
does not contain `-e`.

> I believe replacing pip install -e .[test] with pip install -e .[develop] will do the trick

That seems unlikely, installing the `develop` extra instead of the `test` extra will just install extra development tools, e.g. prospector, as defined here:
https://github.com/ESMValGroup/ESMValCore/blob/c0cc45fde40e4b4f943cb47df262a6d183128a93/setup.py#L211-L213
that shouldn't change the test results.

**Comment by valeriupredoi** [comment]
Does the container need refreshing then? Bouwe, this is a rather serious issue, something that we need to get to the bottom of, otherwise we risk introducing bugs that get uncovered only after merging into `master` and the GA test fails

**Comment by bouweandela** [comment]
The docker container is just 4 days old, you can see it on dockerhub. It will be rebuilt from `master` every time a pull request is merged into it.

**Comment by valeriupredoi** [comment]
OK so indeed, installing it with pip does not pick up the release but rather *some* current branch, but this [test](https://app.circleci.com/pipelines/github/ESMValGroup/ESMValCore/4595/workflows/6ef3855c-8cb6-4551-b414-a82c50be2698/jobs/21570) keeps failing where it should **not** fail if the test is run with the current development branch, is it because `test` installs `master` rather than the development branch where work is done?

**Comment by valeriupredoi** [comment]
the mystery related to this issue is solved by figuring out #1159 - a problem that's got its own isue, so closing this, my apologies for doubting the sanity of your test container @bouweandela :grin:
